casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -20,10 +20,33 @@ import coop.rchain.rspace.{Blake2b256Hash, ReplayException}
 import scala.collection.immutable
 import scala.concurrent.ExecutionContext
 
-class RuntimeManager[F[_]: Concurrent] private (
+abstract class RuntimeManager[F[_]: Concurrent] {
```

Comment:
why not a trait?

Reply:
I like the `[F[_]: Concurrent]` notation and traits don't accept it.
AFAIK an abstract class does not differ from a trait in this context.

Reply:
Out of curiosity - if there exists only one implementation of it why bother turning it into an abstract class?

Reply:
as stated in the blurb: it's for https://github.com/rchain/rchain/pull/2097